### ISSUE TITLE
space-before-function-parentheses should check class constructors

### ISSUE THREAD

**Issue by feross** [eslint/eslint#1989]
eslint incorrectly reports an error for this code:

```js
class Album {
  constructor (name, artist, path) {
    this.name = name
    this.artist = artist
    this.path = path
    this.pictures = []
  }
}

module.exports = Album
```

Line 2: Missing space before function parentheses.

With this config:

```json
{
  "space-before-function-parentheses": [2, "always"]
}
```

Related to #1920.

**Comment by nzakas** [comment]
Likely also fails for concise object literal methods as well.

**Comment by nzakas** [comment]
Working on this.
